ot/lp/_barycenter_solvers.py

```diff
@@ -426,3 +426,172 @@ def generalized_free_support_barycenter(
         return Y, log_dict
     else:
         return Y
+
+
+class StoppingCriterionReached(Exception):
```

Comment:
not a fan of teh exception, because of the try there nuerical error in teh loop might not be detected... could you do a classical keak with variable stopingcriterion please?